bench/wtperf/wtperf.c

```diff
@@ -762,6 +762,90 @@ worker(void *arg)
 					value_buf[0] = 'a';
 				if (opts->random_value)
 					randomize_value(thread, value_buf);
+
+				/*
+				 * Distribute the modifications across
+				 * whole document
+				 */
+				if (opts->distribute_modify)
+				{
+					int nentries = 16;
+					size_t iter;
+					size_t total_modify_size;
+					size_t modify_size;
+					size_t modify_offset;
+
+					/* 10% of total size of the record */
+					total_modify_size = strlen(value) / 10;
+
+					/*
+					 * maximum modification size per
+					 * modify
+					 */
+					modify_size = (total_modify_size
+							/ (size_t)nentries);
+
+					/*
+					 * offset location difference between
+					 * modifications
+					 */
+					modify_offset = (strlen(value)
+							/ (size_t)nentries);
+
+					for (iter = (size_t)nentries;
+						iter > 0; iter--)
+						strncpy(
+						&value_buf[
+						    (iter * modify_offset)
+						    - modify_offset],
+						&value_buf[
+						    (iter * modify_offset)
+						    - modify_size],
+						 modify_size);
+
+					/*
+					 * Induce the modify function overhead
+					 * if configured
+					 */
+					if (opts->induce_modify_overhead)
+					{
+						WT_MODIFY entries[16];
+						WT_ITEM oldv, newv;
+
+						oldv.data = value;
+						oldv.size = strlen(value);
+
+						newv.data = value_buf;
+						newv.size = strlen(value_buf);
+
+						/*
+						 * Reduce the number of
+						 * modifications count, so
+						 * that modify function fails.
+						 */
+						nentries--;
+
+						/*
+						 * Pass the old and new data
+						 * to find out the modify
+						 * vectors, according to the
+						 * passed input, this function
+						 * fail as the modifications
+						 * count reaches the maximum
+						 * number of modifications
+						 * that are allowed for MODIFY
+						 * operation.
```

Comment:
Why do we care that the calc_modify function fails?  If we need it to fail, it would be good to list that in the comment.  If we don't need it to fail, perhaps let it succeed?

I assume we are just calling it to incur the performance overhead of the call?  If that's the case, it would be good to just mention that in the comment.

Reply:
I should have read the ticket first... I have a slightly better understanding of wanting a failure case (off by one).  I still think the code should explain this.

Reply:
I changed the approach of the variable with the new workload option.